State state
Action action
---
State next_state
